bulk_insert.go

```diff
@@ -25,79 +25,58 @@ import (
 func BulkInsert(db *gorm.DB, objects []interface{}, chunkSize int, excludeColumns ...string) error {
 	// Split records with specified size not to exceed Database parameter limit
 	for _, objSet := range splitObjects(objects, chunkSize) {
-		if err := insertObjSet(db, objSet, excludeColumns...); err != nil {
+		_, err := insertObjSet(db, objSet, excludeColumns...)
+		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// BulkInsertWithAssigningIDs executes the query to insert multiple records at once.
-// it will scan the result of `returning id` or `returning *` to [returnedValue] after every insert.
-// it's necessary to set "gorm:insert_option"="returning id" in *gorm.DB
+// BulkInsertWithReturningValues executes the query to insert multiple records at once.
+// This will scan the returned values into `returnedVals`.
 //
-// [returnedValue] slice of primary_key or model, must be a *[]uint(for integer), *[]string(for uuid), *[]struct(for `returning *`)
+// [db] must be set with "gorm:insert_option" to execute RETURNING clause. e.g. db.Set("gorm:insert_option", "RETURNING id")
```

Comment:
how about check using `db.Get("gorm:insert_option")` then warn it to avoid the confusing empty reply?